### PR TITLE
add "variadic" support

### PR DESCRIPTION
**Pull request by xXx-caillou-xXx**
This is _really_ experimental. it seems to work, but i'm not sure that we want that in master.

### CODE REVIEW

File: src/sp_disabled_functions.c [hunk 1]
```diff
@@ -141,8 +147,18 @@ static bool is_param_matching(zend_execute_data* execute_data,
...
       /* This is the parameter name we're looking for. */
       if (true == pcre_matching || config_node->pos != -1) {
+        // Use a local zval to not have to alloc one.
         arg_value = ZEND_CALL_ARG(execute_data, i + 1);
 
+        if (is_variadic) {
+          // This is where the fun begins.
+          // PHP copy the extra arguments in `zend_copy_extra_args` using this delta.
+          // The current argument is never used again.
+          size_t delta = (execute_data->func->op_array.last_var
```
Comment (jvoisin) [inline src/sp_disabled_functions.c]: Wow, this is ugly as sin! PHP does it the same way?

Comment (xXx-caillou-xXx) [reply src/sp_disabled_functions.c]: [Yes](https://github.com/php/php-src/blob/master/Zend/zend_execute.c#L2424), I have no idea why :>

Comment (jvoisin) [reply src/sp_disabled_functions.c]: Ugh.

**Comment by xXx-caillou-xXx** [comment]
Do we want to keep [this](https://github.com/nbs-system/snuffleupagus/pull/238/files#diff-ab2a6a9bc0701513c9a68025a37fe044R26) warning?

**Comment by jvoisin** [comment]
Well, if there is less than N number of parameter passed on a function with a `pos(N)`-hook, we do want to spit a warning, don't we?

**Comment by xXx-caillou-xXx** [comment]
yes, but patching a variadic function with that will result in a lot of useless logs.

**Comment by jvoisin** [comment]
Not that useless, because it'll indicate that something is wrong: if its matching other callsites with a different number of parameters, something is wrong, in my opinion.

**Comment by jvoisin** [comment]
@buixor @kkadosh any opinions about this?

**Comment by jvoisin** [comment]
I think that the rocket-jump-factor is waaaaaaaay too high for this features, and that users will shoot themselves in the foot every single time.

**Pull request closed without merging**